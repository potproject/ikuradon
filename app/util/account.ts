import { sns }  from "../constants/sns";

export function accountURLMigrate(sns: sns, domain: string, url: string){
    // Misskey Migrate Support
    if (sns === "misskey" && url.indexOf("http") !== 0){
        if (url.indexOf("@") !== -1){
            const urlSplit = url.split("@");
            return urlSplit.length === 2 ? `https://${urlSplit[1]}/@${urlSplit[0]}` : url;
        }
        return `https://${domain}/@${url}`;
    }
    return url;
}

export function urlMigrate(sns: sns, domain: string, url: string, tootid: string){
    if (sns === "misskey" && url === ""){
        return `https://${domain}/notes/${tootid}`;
    }
    return url;
}