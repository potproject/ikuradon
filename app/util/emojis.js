import * as CONST_API from "../constants/api";
import Networking from "../services/Networking";

export async function getEmojis(domain){
    try {
        let { data:emojis } = await Networking.fetch(domain, CONST_API.GET_CUSTOMEMOJIS);
        return { emojis, error: null };
    } catch (e){
        return { emojis: [], error: e.message };
    }
}