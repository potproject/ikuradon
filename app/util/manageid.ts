export function getMinMaxId(lowest, highest, data) {
    lowest = lowest === null ? Number.POSITIVE_INFINITY : lowest;
    highest = highest === null ? Number.NEGATIVE_INFINITY : highest;
    let tmp;
    for (let i = data.length - 1; i >= 0; i--) {
        tmp = data[i].id;
        if (tmp < lowest) {
            lowest = tmp;
        }
        if (tmp > highest) {
            highest = tmp;
        }
    }
    return {
        minId: lowest,
        maxId: highest
    };
}
