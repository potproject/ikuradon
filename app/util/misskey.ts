// Misskey Migrate Support Util

export function accountURLMigrate(url: string){
    if (url.indexOf("@") !== -1){
        const urlSplit = url.split("@");
        return urlSplit.length === 2 ? `https://${urlSplit[1]}/@${urlSplit[0]}` : url;
    }
    return url;
}