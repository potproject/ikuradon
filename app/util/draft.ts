import * as Storage from "../util/storage";
import * as CONST_Storage from "../constants/storage";

export async function getDraft(index, deleting = true){
    const draft = await Storage.getItem(CONST_Storage.Draft);
    if (draft === null){
        return "";
    }
    if (typeof draft[index] !== "object" || typeof draft[index].text !== "string"){
        return "";
    }
    if (deleting){
        await deleteDraft(index);
    }
    return draft[index].text;
}

export async function getDraftAll(){
    return await Storage.getItem(CONST_Storage.Draft);
}

export async function addDraft(text){
    let draft = await Storage.getItem(CONST_Storage.Draft);
    if (draft === null){
        draft = [];
    }
    const length = draft.unshift({
        id: Math.random().toString(36).slice(-8),
        text,
    });
    await Storage.setItem(CONST_Storage.Draft, draft);
    // indexを返す
    return length - 1;
}

export async function deleteDraft(index){
    const draft = await Storage.getItem(CONST_Storage.Draft);
    draft.splice(index, 1);
    await Storage.setItem(CONST_Storage.Draft, draft);
    return;
}