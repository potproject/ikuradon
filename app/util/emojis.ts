import * as Rest from "../services/api/Rest";
import * as Session from "../util/session";

export async function getEmojis() {
    try {
        let { sns, domain } = await Session.getDomainAndToken();
        const emojis = await Rest.getInstanceCustomEmojis(sns, domain);
        return { emojis, error: null };
    } catch (e) {
        return { emojis: [], error: e.message };
    }
}

export function getDefaultReaction() {
    const defaultEmoji = ["👍", "❤️", "😆", "🤔", "😮", "🎉", "💢", "😥", "😇", "🍮"];
    return defaultEmoji.map((emoji) => {
        return {
            shortcode: emoji,
            static_url: "",
            url: "",
            visible_in_picker: false
        };
    });
}
