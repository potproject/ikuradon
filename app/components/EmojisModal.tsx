import React, { useEffect, useState } from "react";
import { View, FlatList, StyleSheet, TouchableOpacity } from "react-native";
import { Image } from "react-native-elements";
import { getEmojis } from "../util/emojis";
  
export default function EmojisModal({ current, onSelect }){
    let [emojis, useEmojis] = useState([]);
    useEffect(() => {
        getEmojis(current.domain).then(({ emojis, error }) => {
            if (error === null){
                useEmojis(emojis);
            }
        });
    }, []);
    return (
        <View style={styles.container}>
            <FlatList
                keyExtractor={data => data.shortcode}
                style={styles.wrapList}
                data={emojis}
                numColumns={8}
                renderItem={({ item }) =>(
                    <TouchableOpacity style={styles.emojiList} onPress={() => onSelect(item.shortcode)}>
                        <Image source={{ uri: item.url }} style={styles.emoji}/>
                    </TouchableOpacity>
                )}
            />
        </View>
    );
}

const styles = StyleSheet.create({
    container:{
        width: 320,
        height: 320,
    },
    wrapList: {
        flexDirection: "column",
    },
    emojiList: {
        width: 40,
        height: 40,
    },
    emoji: {
        width:32,
        height:32,
    }
});