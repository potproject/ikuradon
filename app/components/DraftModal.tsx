import React, { useEffect, useState } from "react";
import { View, FlatList, StyleSheet } from "react-native";
import { ListItem } from "react-native-elements";
import { getDraftAll } from "../util/draft";
  
export default function DraftModal({ onSelect }){
    let [drafts, useDrafts] = useState([]);
    useEffect(() => {
        getDraftAll().then((drafts) => useDrafts(drafts));
    }, []);
    return (
        <View style={styles.container}>
            <FlatList
                style={styles.wrapList}
                data={drafts}
                renderItem={({ item }) =>(
                    <ListItem
                        key={item.id}
                        bottomDivider
                        onPress={() => onSelect(item.id, item.text)}
                    >
                        <ListItem.Content>
                            <ListItem.Title>{item.text}</ListItem.Title>
                        </ListItem.Content>
                        <ListItem.Chevron />
                    </ListItem>
                )}
            />
        </View>
    );
}

const styles = StyleSheet.create({
    container:{
        width: 320,
        height: 320,
    }
});