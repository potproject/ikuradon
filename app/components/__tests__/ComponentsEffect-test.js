import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react-native";

import DraftModal from "../DraftModal";
import EmojisModal from "../EmojisModal";
import { getDraftAll } from "../../util/draft";
import { getEmojis } from "../../util/emojis";

import ExampleCurrent from "../../example/current";
import ExampleEmojis from "../../example/emojis";

jest.mock("../../util/draft");
jest.mock("../../util/emojis");

it("<DraftModal />", async () => {
    getDraftAll.mockImplementation(() => {
        return new Promise((resolve) => {
            resolve([{ id:"wgx9stqc", text:"test" }, { id:"a7vy660a", text:"test2" }]);
        });
    });
    const result = render(
        <DraftModal onSelect={(s)=>null} />
    );

    await waitFor(() => {});
    expect(result).toMatchSnapshot();
});

it("<EmojisModal />", async () => {
    getEmojis.mockImplementation(() => {
        return new Promise((resolve) => {
            resolve({ emojis:ExampleEmojis(), error: null });
        });
    });
    const result = render(
        <EmojisModal current={ExampleCurrent()} onSelect={(s)=>null} />
    );

    await waitFor(() => {});
    expect(result).toMatchSnapshot();
});