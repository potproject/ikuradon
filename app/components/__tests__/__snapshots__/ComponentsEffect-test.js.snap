// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DraftModal /> 1`] = `
<View
  style={
    Object {
      "height": 320,
      "width": 320,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": "wgx9stqc",
          "text": "test",
        },
        Object {
          "id": "a7vy660a",
          "text": "test2",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 0.5,
                "borderColor": "#bcbbc1",
                "flexDirection": "row",
                "padding": 14,
              }
            }
            testID="padView"
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "fontSize": 17,
                  }
                }
                testID="listItemTitle"
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
              >
                test
              </Text>
            </View>
            <View
              style={
                Object {
                  "paddingLeft": 16,
                }
              }
            />
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#D1D1D6",
                          "fontSize": 16,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="iconIcon"
                  >
                    Ôèë
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 0.5,
                "borderColor": "#bcbbc1",
                "flexDirection": "row",
                "padding": 14,
              }
            }
            testID="padView"
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "fontSize": 17,
                  }
                }
                testID="listItemTitle"
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
              >
                test2
              </Text>
            </View>
            <View
              style={
                Object {
                  "paddingLeft": 16,
                }
              }
            />
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#D1D1D6",
                          "fontSize": 16,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="iconIcon"
                  >
                    Ôèë
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;

exports[`<EmojisModal /> 1`] = `
<View
  style={
    Object {
      "height": 320,
      "width": 320,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "shortcode": "ms_rainbow_flag",
          "static_url": "https://files.mastodon.social/custom_emojis/images/000/028/691/static/6de008d6281f4f59.png",
          "url": "https://files.mastodon.social/custom_emojis/images/000/028/691/original/6de008d6281f4f59.png",
          "visible_in_picker": true,
        },
        Object {
          "shortcode": "ms_bisexual_flag",
          "static_url": "https://files.mastodon.social/custom_emojis/images/000/050/744/static/02f94a5fca7eaf78.png",
          "url": "https://files.mastodon.social/custom_emojis/images/000/050/744/original/02f94a5fca7eaf78.png",
          "visible_in_picker": true,
        },
        Object {
          "shortcode": "ms_nonbinary_flag",
          "static_url": "https://files.mastodon.social/custom_emojis/images/000/105/099/static/8106088bd4782072.png",
          "url": "https://files.mastodon.social/custom_emojis/images/000/105/099/original/8106088bd4782072.png",
          "visible_in_picker": true,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flexDirection": "column",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 40,
                "opacity": 1,
                "width": 40,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <Image
                onLoad={[Function]}
                source={
                  Object {
                    "uri": "https://files.mastodon.social/custom_emojis/images/000/028/691/original/6de008d6281f4f59.png",
                  }
                }
                style={
                  Object {
                    "bottom": 0,
                    "height": 32,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 32,
                  }
                }
                testID="RNE__Image"
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                transition={true}
              />
              <View
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#bdbdbd",
                      "height": 32,
                      "justifyContent": "center",
                      "width": 32,
                    }
                  }
                  testID="RNE__Image__placeholder"
                />
              </View>
              <View
                style={
                  Object {
                    "height": 32,
                    "width": 32,
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 40,
                "opacity": 1,
                "width": 40,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <Image
                onLoad={[Function]}
                source={
                  Object {
                    "uri": "https://files.mastodon.social/custom_emojis/images/000/050/744/original/02f94a5fca7eaf78.png",
                  }
                }
                style={
                  Object {
                    "bottom": 0,
                    "height": 32,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 32,
                  }
                }
                testID="RNE__Image"
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                transition={true}
              />
              <View
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#bdbdbd",
                      "height": 32,
                      "justifyContent": "center",
                      "width": 32,
                    }
                  }
                  testID="RNE__Image__placeholder"
                />
              </View>
              <View
                style={
                  Object {
                    "height": 32,
                    "width": 32,
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 40,
                "opacity": 1,
                "width": 40,
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <Image
                onLoad={[Function]}
                source={
                  Object {
                    "uri": "https://files.mastodon.social/custom_emojis/images/000/105/099/original/8106088bd4782072.png",
                  }
                }
                style={
                  Object {
                    "bottom": 0,
                    "height": 32,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 32,
                  }
                }
                testID="RNE__Image"
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                transition={true}
              />
              <View
                accessibilityElementsHidden={true}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#bdbdbd",
                      "height": 32,
                      "justifyContent": "center",
                      "width": 32,
                    }
                  }
                  testID="RNE__Image__placeholder"
                />
              </View>
              <View
                style={
                  Object {
                    "height": 32,
                    "width": 32,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`<EmojisModal /> Network Error 1`] = `
<View
  style={
    Object {
      "height": 320,
      "width": 320,
    }
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flexDirection": "column",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`<UserList /> 1`] = `
<View
  style={Object {}}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomWidth": 0.5,
          "borderColor": undefined,
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "height": 40,
            "width": 40,
          }
        }
        theme={
          Object {
            "colors": Object {
              "grey0": "#657786",
              "grey1": "#2b90d9",
              "grey2": "#8899a6",
              "platform": Object {
                "android": Object {},
                "ios": Object {},
              },
              "primary": "#FF3300",
            },
            "customColors": Object {
              "char": "#000000",
              "charBackground": "#FFFFFF",
              "charReverse": "#FFFFFF",
              "item": Object {
                "bookmark": "#4c6ef5",
                "boost": "#2b90d9",
                "favourite": "#ffd27d",
                "none": "#8899a6",
              },
              "link": "#1b95e0",
              "primaryBackground": "#FFFFFF",
              "primaryComplementary": "#00CCFF",
              "tootButton": "#FF5500",
            },
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 20,
              "flex": 1,
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <Image
            borderRadius={20}
            onLoad={[Function]}
            source={
              Object {
                "uri": "https://files.mastodon.social/accounts/avatars/000/023/634/original/6ca8804dc46800ad.png",
              }
            }
            style={
              Object {
                "bottom": 0,
                "flex": 1,
                "height": null,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": null,
              }
            }
            testID="RNE__Image"
            theme={
              Object {
                "colors": Object {
                  "grey0": "#657786",
                  "grey1": "#2b90d9",
                  "grey2": "#8899a6",
                  "platform": Object {
                    "android": Object {},
                    "ios": Object {},
                  },
                  "primary": "#FF3300",
                },
                "customColors": Object {
                  "char": "#000000",
                  "charBackground": "#FFFFFF",
                  "charReverse": "#FFFFFF",
                  "item": Object {
                    "bookmark": "#4c6ef5",
                    "boost": "#2b90d9",
                    "favourite": "#ffd27d",
                    "none": "#8899a6",
                  },
                  "link": "#1b95e0",
                  "primaryBackground": "#FFFFFF",
                  "primaryComplementary": "#00CCFF",
                  "tootButton": "#FF5500",
                },
              }
            }
            transition={true}
          />
          <View
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#bdbdbd",
                  "flex": 1,
                  "height": null,
                  "justifyContent": "center",
                  "width": null,
                }
              }
              testID="RNE__Image__placeholder"
            />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "height": null,
                "width": null,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "paddingLeft": 16,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#000000",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
          theme={
            Object {
              "colors": Object {
                "grey0": "#657786",
                "grey1": "#2b90d9",
                "grey2": "#8899a6",
                "platform": Object {
                  "android": Object {},
                  "ios": Object {},
                },
                "primary": "#FF3300",
              },
              "customColors": Object {
                "char": "#000000",
                "charBackground": "#FFFFFF",
                "charReverse": "#FFFFFF",
                "item": Object {
                  "bookmark": "#4c6ef5",
                  "boost": "#2b90d9",
                  "favourite": "#ffd27d",
                  "none": "#8899a6",
                },
                "link": "#1b95e0",
                "primaryBackground": "#FFFFFF",
                "primaryComplementary": "#00CCFF",
                "tootButton": "#FF5500",
              },
            }
          }
        >
          noiob
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingLeft": 16,
          }
        }
      />
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#2b90d9",
                    "fontSize": 24,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "material",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              Óóâ
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
